x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/process/MlMemoryTracker.java

```diff
@@ -389,7 +396,7 @@ private void refreshAllDataFrameAnalyticsJobTasks(List<PersistentTasksCustomMeta
      */
     public void refreshAnomalyDetectorJobMemory(String jobId, ActionListener<Long> listener) {
         if (isMaster == false) {
-            listener.onResponse(null);
+            listener.onFailure(new NotMasterException("Request to refresh anomaly detector memory requirement on non-master node"));
```

Comment:
This pattern makes more sense to me but it will affect the behaviour of the open job and upgrade model snapshot actions which would have continued with the null response before. Both of those actions are master node actions so the change will only be seen if the node stops being master when during processing the action and that would be a very difficult problem to solve. We will have to see what CI makes of the change.

Reply:
> change will only be seen if the node stops being master when during processing the action and that would be a very difficult problem to solve.

True, but a node only stops being master if it dies. Do nodes stop being a master node for any other reason than crashing?

Reply:
> True, but a node only stops being master if it dies. Do nodes stop being a master node for any other reason than crashing?

Yes, there is proof of this happening in https://github.com/elastic/elasticsearch/issues/68685#issuecomment-809546734.  `node_t1` becomes master, then ceases to be master 119ms later, then becomes master again 155ms after that.

I am going to mention this in the team meeting today because I suspect there are other places where we haven't coded for this possibility.  Any class that reacts to going on and off master, and has multiple accessor methods that we call from the same operation will have a problem _if_ the calling code is making assumptions that the two accessors called were both called on master or off master. 😬